kernel/comps/softirq/src/lib.rs

```diff
@@ -119,25 +127,87 @@ static ENABLED_MASK: AtomicU8 = AtomicU8::new(0);
 
 cpu_local_cell! {
     static PENDING_MASK: u8 = 0;
+    static DISABLE_SOFTIRQ_COUNT: u8 = 0;
+}
+
+fn increase_disable_softirq_count() {
```

Comment:
I think it's better to:
 - move `DISABLE_SOFTIRQ_COUNT` to a standalone module (e.g., `lock` module?), and
 - remove all the single-line wrappers (they give the reader a sense that you're trying to build some abstractions, but there's no abstraction really being built; for example, the caller still has to make sure that preemption is disabled itself).